### ISSUE TITLE
Make all value+unit of measurement uniform and correct

### ISSUE THREAD

**Issue by Andrew15-5** [typst/typst#3220]
### Description

Currently, some units in the compiler's output are written with whitespace after the value, and others are smashed together with the value. The correct way is to leave a single whitespace between a value and its unit of measurement. The information size and transfer speed measurements do have whitespace.

Yes, a lot of apps skip the whitespace, but this is now what we were taught to do (in school); yes, it "looks better" without a whitespace, but this is only because it became "normal", if we use space everywhere, then it also will become "normal" and will "look better". See #2100 for more info.

I have found 2 places where units of time don't have a whitespace on the left side/prepended:

https://github.com/typst/typst/blob/6ac71eeaf7b68dab07f75bd1a480810481fa9b73/crates/typst-cli/src/download.rs#L217-L235

https://github.com/typst/typst/blob/6ac71eeaf7b68dab07f75bd1a480810481fa9b73/crates/typst-cli/src/watch.rs#L189-L192

But in order to fix this, I wanted to ask do we _really_ need "days" and "hours" (or even "minutes") for download times? For compile times, we only get seconds and milliseconds. Also, for downloads, the time format is manual (can be fixed with "1 char"), but for compilations, the time is auto-generated by using debug trait for `std::time::Duration`. I thought maybe we can uniform the both and use the same formatter function. Though everything looks localized and there is no `global_util_functions.rs` file, so I also don't really want to copy the same code twice, though "twice" is not critical especially if the function will be very small.

P.S. I know that it was implemented by different people.

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
We could consider moving the time formatting to `typst-utils`.

**Comment by Andrew15-5** [comment]
I've started messing around with this. And the first problem is that those 2 places use/expect different output formats. So we probably need 2 separate functions. How about `format::eta_time` and `format::compilation_time`? I probably will combine the two internally so that the code stays DRY. For the second one, I created `pub time_as_ms` and `time_as_ms_with_precision`.

P.S. I started testing by just compiling a document, but then quickly realized that it's super unproductive. So I create a test for the `time_as_ms_with_precision`. Now the compilation is fast and everything is configurable (output, input, precision).
